---
title: "Narrative"
output: html_document
---

```{r load packages, echo=TRUE, message=FALSE, warning=FALSE}
library(tidyverse)
library(methylKit)
```

# Introduction

It is clear environmental conditions have an influence on bivalvs

To better understand the mechanisms, oyster from a single location were grown in two separate enviroments for 26 months and methylated differences assessed.

This design allows us to see the influence of conditions on DNA methylation that is not necessarily genome dependent as from one population.

# Methods

## *Oysters*

Adult Olympia oyster originally from Fidalgo Bay, in North Puget Sound (Washington, US) were grown out at two locations including Fidalgo Bay and Oyster Bay (located in South Puget Sound). Oysters used in this study were part of a larger experiment described elsewhere (Heare et al 2017). Oysters were outplanted in August 2013 and collected in November 2015.

## *Bisulfite treated DNA library preparation*

At the end of grow out period, genomic DNA was isolated from whole bodies preserved in RNAlater (Ambion) from four individuals grown in Fidalgo Bay and four individuals grown in Oyster Bay with DNAzol (Molecular Research, Inc.). Tissue was homogenized with disposable mortar and pestle in 500uL of DNAzol in a 1.7mL snap cap tube. An additional 500uL of DNAzol was added, along with 10uL of RNase A (10mg/mL; ThermoFisher), mixed by inversion, and incubated at room temperature (RT) for 10 minutes. Chloroform (300uL) was added, mixed, incubated at RT for 5 minutes, then centrifuged for 10 mins at 12,000g at RT. The aqueous phase was transferred to a clean tube 1.7mL snap cap tube and subjected to a standard ethanol precipitation. The precipitated DNA was resuspended in 100uL of Buffer EB (Qiagen). Samples were quantified on a Qubit 3.0 (ThermoFisher) using the Qubit dsDNA BR assay (Invitrogen). Sample integrity was verified via agarose gel.

Genomic DNA was subjected to bisulfite conversion using the EZ DNA Methylation-Gold Kit (ZymoResearch) according to the manufacturer's protocol, with the following changes; 100ng of DNA was used, centrifugation steps were performed at 13,000g, and desulphonation incubation period was 20 minutes.

Sequencing libraries were constructed using the TruSeq DNA Methylation Library Kit (Illumina) according to the manufacturer's protcool. Each sample received a unique barcode. Each library was quantified using a Qubit 3.0 and the Qubit dsDNA HS assay (Invitrogen). A single pool, using 10ng of each library, was prepared for sequencing.

Sequencing was performed in a single flowcell on a HiSeq2500 (Illumina) with 50bp single-end reads. Raw sequence reads were deposited in the NCBI Short Read Archive (NCBI SRA: SRP163248). Read quality was evaluated with FastQC (v0.11.7; <https://www.bioinformatics.babraham.ac.uk/projects/fastqc/>) and visualized with MultiQC (v1.6.dev0; <http://multiqc.info/>).

## *Draft Genome Assembly and Annotation*

A draft genome was generated to map bisulfite treated sequence reads. The draft genome for the Olympia oyster was created using a combination of short read sequence data (Illumina HiSeq4000) combined with long read sequence data (PacBio RSII) using PBJelly (PBSuite_15.8.24; English et al, 2012). Short reads (NCBI SRA: SRP072461) were assembled using SOAPdenovo (Li et al, 2008). The scaffolds (n=765,755) from this assembly were combined with the PacBio long read data (NCBI SRA: SRR5809355) using PBJelly (PBSuite_15.8.24; English et al, 2012). Assembly with PBJelly was performed using the default settings. See Supplemental File \#1 for code used to run assembly. Only contigs longer than 1000 bp were used for further analysis. Genome assembly parameters were compiled using QUAST (v4.5; Gurevich et al, 2013).

Genome annotation was performed using MAKER (v.2.31.10; Campbell et al, 2014) configured to use with Message Passing Interface (MPI). A custom repeat library (Ostrea_lurida_v081-families.fa) for use in MAKER was generated using RepeatModeler (open-1.0.11; <https://github.com/rmhubley/RepeatModeler/releases/tag/open-1.0.11>). RepeatModeler was configured with the following softare: RepeatMasker (open-4.0.7; <http://www.repeatmasker.org/RMDownload.html>) configured with Repbase RepeatMasker (v20170127; <https://www.girinst.org/server/RepBase/index.php>), RECON (v1.08; Bao and Eddy, 2002) with RepeatMasker patch, RepeatScout (v1.0.5; Price et al, 2005), RMBLast (2.6.0; <http://www.repeatmasker.org/RMBlast.html>) configured with the isb-2.6.0+-changes-vers2 patch file, and TRF (v4.0.4; Benson, 1999).

MAKER was run on two high performance computing (HPC) nodes (Lenov NextScale, E5-2680 v4 dual CPUs, 28 cores, 128GB RAM) on the University of Washington's shared scalable compute cluster (Hyak) using the icc_19-ompi_3.1.2 module (Intel C compiler v19, Open MPI v3.1.2). An Olympia oyster transcriptome assembly produced by our lab (available here: <http://eagle.fish.washington.edu/cnidarian/Olurida_transcriptome_v3.fasta>) was used for EST data. Protein data used was a concatenation of NCBI proteomes from *Crassostrea gigas* (GCA_000297895.1_oyster_v9_protein.faa) and *Crassostrea virginica* (GCF_002022765.2_C\_virginica-3.0_protein.faa). *Ab-initio* gene training was performed twice using the included SNAP software (Korf, 2004). Functional protein annotation was performed using BLASTp (v.2.6.0+; Altschul et al, 1990) against a UniProt BLAST database (FastA file formatted using BLAST 2.8.1+) downloaded on 01/09/2019. The MAKER functions `maker_functional_gff` and `maker_functional_fasta` both used this same UniProt BLAST database. Protein domain annotation was performed using InterProScan 5(v5.31-70.0; Jones et al, 2014). See Supplemental File \#1 for code used for repeat library creation and genome annotation.

## General Methylation Landscape

In order to describe the general DNA methylation landscape the entirety of the data was combined and mapped to the draft genome. In preparation for bisufite-treated sequence mapping using Bismark (v0.22.0; [\<https://github.com/FelixKrueger/Bismark\>](https://github.com/FelixKrueger/Bismark){.uri}), 10bp from the 5' end of each read were trimmed using TrimGalore! (v0.4.5; [\<https://www.bioinformatics.babraham.ac.uk/projects/trim_galore/\>](https://www.bioinformatics.babraham.ac.uk/projects/trim_galore/){.uri}). Post-trimming read quality was assessed with FastQC (v0.11.9; [\<https://www.bioinformatics.babraham.ac.uk/projects/fastqc/\>](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/){.uri}) and visualized with MultiQC (v1.6.dev0;[\<http://multiqc.info/\>](http://multiqc.info/){.uri}).

```{bash catBS, eval = FALSE, echo = FALSE}
#!/bin/bash
## Job Name
#SBATCH --job-name=olycatC
## Allocation Definition
#SBATCH --account=coenv
#SBATCH --partition=coenv
## Nodes
#SBATCH --nodes=1
## Walltime (days-hours:minutes:seconds format)
#SBATCH --time=5-00:00:00
## Memory per node
#SBATCH --mem=100G
#SBATCH --mail-type=ALL
#SBATCH --mail-user=sr320@uw.edu
## Specify the working directory for this job
#SBATCH --chdir=/gscratch/scrubbed/sr320/061421-olyC



# Directories and programs
bismark_dir="/gscratch/srlab/programs/Bismark-0.22.3"
bowtie2_dir="/gscratch/srlab/programs/bowtie2-2.3.4.1-linux-x86_64/"
samtools="/gscratch/srlab/programs/samtools-1.9/samtools"
reads_dir="/gscratch/srlab/sr320/data/olurida-bs/"
fastqc="/gscratch/srlab/programs/fastqc_v0.11.9/fastqc"
genome_folder="/gscratch/srlab/sr320/data/olurida-bs/genome/"

source /gscratch/srlab/programs/scripts/paths.sh


# 
# ${bismark_dir}/bismark_genome_preparation \
# --verbose \
# --parallel 28 \
# --path_to_aligner ${bowtie2_dir} \
# ${genome_folder}
# 




#FastQC 

#Populate array with FastQ files
#fastq_array=(/gscratch/srlab/sr320/data/cg-big/*.fq.gz)

#Pass array contents to new variable
#fastqc_list=$(echo "${fastq_array[*]}")

#Run FastQC
# ${fastqc} \
# --threads 28 \
# /gscratch/srlab/sr320/data/olurida-bs/combined.fq.gz \
# -o /gscratch/scrubbed/sr320/061021-oly/





#1_ATCACG_L001_R1_001_trimmed.fq.gz

find ${reads_dir}combi* \
| xargs basename -s .fq.gz | xargs -I{} ${bismark_dir}/bismark \
--path_to_bowtie ${bowtie2_dir} \
-genome ${genome_folder} \
-p 4 \
-score_min L,0,-0.6 \
${reads_dir}{}.fq.gz  


find *.bam | \
xargs basename -s .bam | \
xargs -I{} ${bismark_dir}/deduplicate_bismark \
--bam \
--single \
{}.bam



${bismark_dir}/bismark_methylation_extractor \
--single-end \
--comprehensive \
--merge_non_CpG \
--bedGraph \
--counts \
--scaffolds \
--multicore 28 \
--buffer_size 75% \
--report \
*deduplicated.bam



# Bismark processing report

${bismark_dir}/bismark2report

#Bismark summary report

${bismark_dir}/bismark2summary





# Sort files for methylkit and IGV

find *deduplicated.bam | \
xargs basename -s .bam | \
xargs -I{} ${samtools} \
sort --threads 28 {}.bam \
-o {}.sorted.bam

# Index sorted files for IGV
# The "-@ 16" below specifies number of CPU threads to use.

find *.sorted.bam | \
xargs basename -s .sorted.bam | \
xargs -I{} ${samtools} \
index -@ 28 {}.sorted.bam


find *deduplicated.bismark.cov.gz \
| xargs basename -s _bismark_bt2.deduplicated.bismark.cov.gz \
| xargs -n 1 -P 28 -I{} ${bismark_dir}/coverage2cytosine \
--genome_folder ${genome_folder} \
-o {} \
--merge_CpG \
--zero_based \
{}_bismark_bt2.deduplicated.bismark.cov.gz

#creating bedgraphs post merge

for f in *merged_CpG_evidence.cov
do
  STEM=$(basename "${f}" .CpG_report.merged_CpG_evidence.cov)
  cat "${f}" | awk -F $'\t' 'BEGIN {OFS = FS} {if ($5+$6 >= 10) {print $1, $2, $3, $4}}' \
  > "${STEM}"_10x.bedgraph
done



for f in *merged_CpG_evidence.cov
do
  STEM=$(basename "${f}" .CpG_report.merged_CpG_evidence.cov)
  cat "${f}" | awk -F $'\t' 'BEGIN {OFS = FS} {if ($5+$6 >= 5) {print $1, $2, $3, $4}}' \
  > "${STEM}"_5x.bedgraph
done


#creating tab files with raw count for glms

for f in *merged_CpG_evidence.cov
do
  STEM=$(basename "${f}" .CpG_report.merged_CpG_evidence.cov)
  cat "${f}" | awk -F $'\t' 'BEGIN {OFS = FS} {if ($5+$6 >= 10) {print $1, $2, $3, $4, $5, $6}}' \
  > "${STEM}"_10x.tab
done


for f in *merged_CpG_evidence.cov
do
  STEM=$(basename "${f}" .CpG_report.merged_CpG_evidence.cov)
  cat "${f}" | awk -F $'\t' 'BEGIN {OFS = FS} {if ($5+$6 >= 5) {print $1, $2, $3, $4, $5, $6}}' \
  > "${STEM}"_5x.tab
done



 #run multiqc
/gscratch/srlab/programs/anaconda3/bin/multiqc .

```

Genome coverage and characterization of methylation landscape was carried out using CpG coverage along with Bedtools and genome annotations files.

```{r readin_catcov, warning=TRUE, include=FALSE, cache=TRUE}
cov <- read.csv("../data_21/combined.CpG_report.merged_CpG_evidence.cov.txt", sep="\t", header = FALSE) %>%
  mutate(cvr = V5 + V6)
```

```{r include=FALSE}
cov5 <- read.csv("../data_21/combined_5x.bedgraph", sep="\t", header = FALSE)
cov10 <- read.csv("../data_21/combined_10x.bedgraph", sep="\t", header = FALSE)
```

## *Differential Methylational analysis*

In order to examine methylation in individual oysters including characterizing differential methylation quality trimming and mapping was performed as described.

```{bash eval=FALSE, include=FALSE}
#!/bin/bash
## Job Name
#SBATCH --job-name=olywgbs
## Allocation Definition
#SBATCH --account=coenv
#SBATCH --partition=coenv
## Nodes
#SBATCH --nodes=1
## Walltime (days-hours:minutes:seconds format)
#SBATCH --time=5-00:00:00
## Memory per node
#SBATCH --mem=100G
#SBATCH --mail-type=ALL
#SBATCH --mail-user=sr320@uw.edu
## Specify the working directory for this job
#SBATCH --chdir=/gscratch/scrubbed/sr320/061421-oly
  
  
  
# Directories and programs
bismark_dir="/gscratch/srlab/programs/Bismark-0.22.3"
bowtie2_dir="/gscratch/srlab/programs/bowtie2-2.3.4.1-linux-x86_64/"
samtools="/gscratch/srlab/programs/samtools-1.9/samtools"
reads_dir="/gscratch/srlab/sr320/data/olurida-bs/"
genome_folder="/gscratch/srlab/sr320/data/olurida-bs/genome/"
  
source /gscratch/srlab/programs/scripts/paths.sh
  
#   
#   
# ${bismark_dir}/bismark_genome_preparation \
# --verbose \
# --parallel 28 \
# --path_to_aligner ${bowtie2_dir} \
# ${genome_folder}
  
  
#FastQC 

#Populate array with FastQ files
fastq_array=(/gscratch/srlab/sr320/data/olurida-bs/*.fq.gz)

#Pass array contents to new variable
fastqc_list=$(echo "${fastq_array[*]}")

#Run FastQC
${fastqc} \
--threads 28 \
${fastqc_list} \
-o /gscratch/scrubbed/sr320/061421-oly/

  
  
  
  
#1_ATCACG_L001_R1_001_trimmed.fq.gz
  
find ${reads_dir}*_R1_001_trimmed.fq.gz \
| xargs basename -s _R1_001_trimmed.fq.gz | xargs -I{} ${bismark_dir}/bismark \
--path_to_bowtie ${bowtie2_dir} \
-genome ${genome_folder} \
-p 4 \
-score_min L,0,-0.6 \
--non_directional \
${reads_dir}{}_R1_001_trimmed.fq.gz  
  
  
find *.bam | \
xargs basename -s .bam | \
xargs -I{} ${bismark_dir}/deduplicate_bismark \
--bam \
--single \
{}.bam



${bismark_dir}/bismark_methylation_extractor \
--single-end \
--comprehensive \
--merge_non_CpG \
--bedGraph \
--counts \
--scaffolds \
--multicore 28 \
--buffer_size 75% \
--report \
*deduplicated.bam
  
  
# Bismark processing report
  
${bismark_dir}/bismark2report
  
#Bismark summary report
  
${bismark_dir}/bismark2summary
  

  
# Sort files for methylkit and IGV
  
find *deduplicated.bam | \
xargs basename -s .bam | \
xargs -I{} ${samtools} \
sort --threads 28 {}.bam \
-o {}.sorted.bam
  
# Index sorted files for IGV
# The "-@ 16" below specifies number of CPU threads to use.
  
find *.sorted.bam | \
xargs basename -s .sorted.bam | \
xargs -I{} ${samtools} \
index -@ 28 {}.sorted.bam
  
  
find *deduplicated.bismark.cov.gz \
| xargs basename -s _L001_R1_001_trimmed_bismark_bt2.deduplicated.bismark.cov.gz \
| xargs -I{} ${bismark_dir}/coverage2cytosine \
--genome_folder ${genome_folder} \
-o {} \
--merge_CpG \
--zero_based \
{}_L001_R1_001_trimmed_bismark_bt2.deduplicated.bismark.cov.gz


#creating bedgraphs post merge

for f in *merged_CpG_evidence.cov
do
  STEM=$(basename "${f}" .CpG_report.merged_CpG_evidence.cov)
  cat "${f}" | awk -F $'\t' 'BEGIN {OFS = FS} {if ($5+$6 >= 10) {print $1, $2, $3, $4}}' \
  > "${STEM}"_10x.bedgraph
done



for f in *merged_CpG_evidence.cov
do
  STEM=$(basename "${f}" .CpG_report.merged_CpG_evidence.cov)
  cat "${f}" | awk -F $'\t' 'BEGIN {OFS = FS} {if ($5+$6 >= 5) {print $1, $2, $3, $4}}' \
  > "${STEM}"_5x.bedgraph
done


#creating tab files with raw count for glms

for f in *merged_CpG_evidence.cov
do
  STEM=$(basename "${f}" .CpG_report.merged_CpG_evidence.cov)
  cat "${f}" | awk -F $'\t' 'BEGIN {OFS = FS} {if ($5+$6 >= 10) {print $1, $2, $3, $4, $5, $6}}' \
  > "${STEM}"_10x.tab
done


for f in *merged_CpG_evidence.cov
do
  STEM=$(basename "${f}" .CpG_report.merged_CpG_evidence.cov)
  cat "${f}" | awk -F $'\t' 'BEGIN {OFS = FS} {if ($5+$6 >= 5) {print $1, $2, $3, $4, $5, $6}}' \
  > "${STEM}"_5x.tab
done

 #run multiqc
/gscratch/srlab/programs/anaconda3/bin/multiqc . 
  

```

Identification of differentially methylated regions (50 bp windows) using R (v3.4.4; R Studio v1.1.383) and the MethylKit R package (v1.4.1; Akalin et al, 2012), with a minimum of 5x coverage. Differnailly methylated regions were defined as those with at least a 25% difference (p \< 0.01) in methylation across oysters grown out at the two locales. All code and analyses is available in the following GitHub repository: URL_NEEDED.

```{bash eval=FALSE, include=FALSE}
wget -r \
--no-check-certificate \
--no-directories --no-parent \
-P ../data_21/ \
-A *sorted.bam \
https://gannet.fish.washington.edu/seashell/bu-mox/scrubbed/061421-oly/
```

```{r include=FALSE}
file.list_21=list('../data_21/1_ATCACG_L001_R1_001_trimmed_bismark_bt2.deduplicated.sorted.bam',
                 '../data_21/2_CGATGT_L001_R1_001_trimmed_bismark_bt2.deduplicated.sorted.bam',
                 '../data_21/3_TTAGGC_L001_R1_001_trimmed_bismark_bt2.deduplicated.sorted.bam',
                 '../data_21/4_TGACCA_L001_R1_001_trimmed_bismark_bt2.deduplicated.sorted.bam',
                 '../data_21/5_ACAGTG_L001_R1_001_trimmed_bismark_bt2.deduplicated.sorted.bam',
                 '../data_21/6_GCCAAT_L001_R1_001_trimmed_bismark_bt2.deduplicated.sorted.bam',
                 '../data_21/7_CAGATC_L001_R1_001_trimmed_bismark_bt2.deduplicated.sorted.bam',
                 '../data_21/8_ACTTGA_L001_R1_001_trimmed_bismark_bt2.deduplicated.sorted.bam'
)
```

```{r eval=FALSE, include=FALSE}
myobj_21 = processBismarkAln(location = file.list_21, sample.id = list("1","2","3","4","5","6","7","8"), assembly = "v081", read.context="CpG", mincov=2, treatment = c(0,0,0,0,1,1,1,1))
```

```{r include=FALSE}
#save(myobj_21, file = "../analyses/myobj_21")

```

```{r include=FALSE}
load("../analyses/myobj_21")
```

```{r message=FALSE, warning=FALSE}
filtered.myobj=filterByCoverage(myobj_21,lo.count=10,lo.perc=NULL,
                                      hi.count=NULL,hi.perc=NULL)

meth_filter=unite(filtered.myobj, min.per.group=2L, destrand=TRUE)

clusterSamples(meth_filter, dist="correlation", method="ward", plot=TRUE)


PCASamples(meth_filter)

```

```{r}
myDiff=calculateDiffMeth(meth_filter,mc.cores=8)

```

```{r}
# get hyper methylated bases
myDiff20p.hyper=getMethylDiff(myDiff,difference=20,qvalue=0.01,type="hyper")
#
# get hypo methylated bases
myDiff20p.hypo=getMethylDiff(myDiff,difference=20,qvalue=0.01,type="hypo")
#
#
# get all differentially methylated bases
myDiff20p=getMethylDiff(myDiff,difference=20,qvalue=0.01)





```

```{r}
write.table(myDiff20p, file = "../analyses/myDiff20p.tab", sep = "\t")
```

-- ADD ANNOTATION METHODS

All of the TrimGalore!, FastQC, MultiQC, and Bismark processing was performed on a compute node (CentOS 7.4.1708) in the University of Washington's high performance computing (HPC) cluster. See Supplemental File \#1 for code.

# Results

*DNA Sequence Data Availablity*

Bisulfite treated DNA sequencing resulted in 11,954,873 average reads per library. The draft genome assembly yielded 213,264 contigs, comprised of 1.2Mbp with an N50 of 12,433. The assembly contained 894 contigs \>= 50,000bp. The largest contig was 194,507bp (Table NUMBER).

--- ADD % MAPPING

## General DNA Methylation Landscape

```{r cg, cache = TRUE, include = FALSE}
#cg <- read.csv("../data_21/Olurida_v081_CG-motif.gff", sep="\t", header = FALSE)
```

```{r methCpG, include=FALSE}
meth <- filter (cov5, V4>=50)
 write.table(meth, file = "../data_21/meth_loc.bed", sep = "\t", col.names = FALSE, row.names = FALSE, quote = FALSE)
lessmeth <- filter (cov5, V4<50)
 write.table(lessmeth, file = "../data_21/lessmeth_loc.bed", sep = "\t", col.names = FALSE, row.names = FALSE, quote = FALSE) 
nrow(meth)+nrow(lessmeth)
```

```{bash methgene, eval=FALSE, include=FALSE}
 /Applications/bioinfo/bedtools2/bin/intersectbed \
 -u \
 -wa \
 -a ../data_21/meth_loc.bed \
 -b ../data_21/Olurida_v081-20190709.gene.gff |
 wc -l
```

```{bash lessmethgene, eval=FALSE, include=FALSE}
/Applications/bioinfo/bedtools2/bin/intersectbed \
 -u \
 -wa \
 -a ../data_21/lessmeth_loc.bed \
 -b ../data_21/Olurida_v081-20190709.gene.gff | wc -l 

```

Of the 27810012 loci in the draft genome `r nrow(cov)` (`r (nrow(cov)/27810012)*100`%) had corresponding mapped reads, with `r nrow(cov5)` loci covered at 5x and `r nrow(cov10)` loci covered at 10x. When considering loci with 5x coverage `r ((188345)/nrow(meth))*100`% methylated loci overlapped with predicted genes while `r ((242856)/nrow(lessmeth))*100`% of less methylated loci overlapped with predicted genes.

```{r coverage_plot, echo=FALSE}
ggplot(cov, aes(x=cvr)) +
   geom_histogram(binwidth = 1) +
  scale_y_log10(labels = scales::comma) +
  xlab("CpG coverage depth") + ylab("count") + 
  geom_vline(xintercept = 5,
                color = "grey", size=0.5) +
  labs(caption = "Figure 1: Coverage depth for each loci on genome where read mapped to genome. Grey line indicates threshold for downstream analyses.") +
  theme(plot.caption = element_text(hjust = 0), #Default is hjust=1
        plot.caption.position =  "plot") #NEW parameter
```

```{r distibution_plot, echo=FALSE}
ggplot(cov5, aes(x=V4)) +
   geom_histogram(bins = 50) +
   coord_cartesian(ylim=c(0, 800000))

```

\`\`\`

*Differential Methylation*

Fifty-three differentially methylated regions were identfied. Of these DMRs, 32 were hypomethylated in oysters grown in Oyster Bay (foreign site), and 21 were hypermethylated in oysters grown in the foreign site.

---How much of the genome is covered, how many CpGs

---How many loci examined - I think 23,478

When using regions (DMRs), 104 regions were identified found on 89 contigs.

Image showing Coverage and mCpG ![igv](https://github.com/sr320/nb-2018/blob/master/O_lurida/analyses/wgbs-igv.png?raw=true)
